### PR TITLE
Explore Page & Sidebar Refactoring

### PR DESCRIPTION
**Pull request by horsemaker**
- Refactored Videos DB to add `views` & `uploadedAt`
- Refactored Categories DB to add `All` category
- Added Categories Context, Provider & custom hook useCategories
- Added Videos Context, Provider & custom hook useVideos
- Added getCategories service
- Added getVideos service
- Added Explore Page
- Added Categories Component
- Added Vertical Video Card Component
- Added nFormatter util
- Refactored Sidebar for breakpoint `950px`

recording

### CODE REVIEW

**Review by vijaytembugade — APPROVED** [review]
Great UI 🔥
Code also looks clean. 

Amazing concept

**Review by MiheerTamkhane — APPROVED** [review]
What an app, Working good, looking good, also the code part is awsome.
readable and understandable . Kudos to this man

File: src/frontend/provider/CombinedProvider.js [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
         <AuthProvider>
-          <SidebarProvider>{children}</SidebarProvider>
+          <SidebarProvider>
+            <VideosProvider>
+              <CategoriesProvider>{children}</CategoriesProvider>
+            </VideosProvider>
+          </SidebarProvider>
         </AuthProvider>
       </ThemeProvider>
     </BrowserRouter>
```
Comment (narharikale) [inline src/frontend/provider/CombinedProvider.js]: great work work here✨

File: src/frontend/utils/formatter-utils/nFormatter.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  var item = lookup
+    .slice()
+    .reverse()
+    .find(function (item) {
+      return num >= item.value;
+    });
+  return item
+    ? (num / item.value).toFixed(digits).replace(rx, "$1") + item.symbol
+    : "0";
+};
```
Comment (narharikale) [inline src/frontend/utils/formatter-utils/nFormatter.js:21]: learned new thing , thanks🤩

**Review by narharikale — APPROVED** [review]
Amazing work  yash ✨ ,  and love the approach how you pay attention to small-small things👌

**Review by t007rushi — APPROVED** [review]
Everything looks good
Nice naming 
Just suggestion to have minimal transitions for flashing sidebar

**Pull request merged**